### ISSUE TITLE
Add ability to move ~/.processing directory

### ISSUE THREAD

**Issue by Addisonbean** [processing/processing#6115]
## Description

Processing uses `~/.processing` for it's data with no way to change that. It would be nice to be able to control the location of that directory so it can follow XDG directory specifications and keep users home directories clean.

## Expected Behavior

Processing should provide an environment variable to control where the processing data directory is found, and fall back to `~/.processing` if not specified.

## Current Behavior

There is no clearly documented way to move `~/.processing` to another location.

## Steps to Reproduce

N/A

## Your Environment

* Processing version: 3.5.4
* Operating System and OS version: Arch Linux (Linux kernel 5.9.3) 
* Other information: `$XDG_CONFIG_HOME` and `$XDG_DATA_HOME` are set and were not used by processing when deciding where to create this directory

## Possible Causes / Solutions

Add an environment variable that can provide an alternate location for `~/.processing`, and fall back to the old location if not specified.

An even better solution would be to default to `~/.config/processing` unless `$XDG_CONFIG_HOME` is set or `~/.processing` already exists.

**Comment by benfry** [comment]
Thanks for the report… I see the `.config` location in use on a couple Linux machines here so that makes sense; is there any documentation on how widespread this is/which systems are likely to expect things there? Is it an XDG thing? Limited to GUI apps? Just looking for any sort of spec/references I can use to make the right call here (and revise my 90s UNIX thinking…)

**Comment by Addisonbean** [comment]
Yes, `~/.config` is part of the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).

Here's an incomplete list of common applications that use XDG directories by default (including `~/.config`): https://wiki.archlinux.org/title/XDG_Base_Directory#Support

XDG isn't limited to any particular kind of program. Notable tools from that list which use XDG directories to determine config locations include `git`, `neovim`, `tmux`, `Android Studio`, `Blender`, `emacs`, `LibreOffice`, `wireshark`, `pulseaudio`, `htop`, and more. The link on the Arch Wiki above has links to bug reports on many of these projects, which shows many users view lack of XDG support as a bug.

I know using XDG directories for config and related files is expected by many Linux users across many distros, but I don't know about the "standard" on macOS. I know many of those apps I listed use `~/.config` for config, while others use `~` or even `~/Library/Application Support`. `~/Library/Application Support` and `~/.config` seem like the two most appropriate options on macOS to me.

**Comment by benfry** [comment]
Perfect, thanks; that's the information I was looking for. 

(Moving this over to the 4.x repository so I can work on it there.)
